storage-node/src/services/webApi/controllers/filesApi.ts

```diff
@@ -99,13 +99,15 @@ export async function uploadFile(
     const queryNodeUrl = res.locals.queryNodeEndpoint
     const workerId = res.locals.workerId
 
-    const [, hash] = await Promise.all([
+    const api = res.locals.api
+    const bagId = parseBagId(uploadRequest.bagId)
+
+    const [, , hash] = await Promise.all([
       verifyBucketId(queryNodeUrl, workerId, uploadRequest.storageBucketId),
+      verifyBagAssignment(api, bagId, uploadRequest.storageBucketId),
```

Comment:
It looks like `RequestData` is incorrectly typed, the `storageBucketId` type was set to `number`, but in `api-spec/openapi.yaml` it's defined as `string`:
```yaml
              properties:
                file:
                  description: Data file
                  type: string
                  format: binary
                dataObjectId:
                  description: Data object runtime ID
                  type: string
                  pattern: '^\d+$' #integer
                storageBucketId:
                  description: Storage bucket ID
                  type: string
                  pattern: '^\d+$' #integer
                bagId:
                  description: Bag ID
                  type: string
```
There may be more mistakes like this. Ideally those types would be just autogenerated from `openapi.yml` spec, I'll make an issue to fix this.

The bottom line is that `verifyBagAssignment` now always fails because it compares `number` bucket ids to a `string` bucket id:
```
Uploading object 0 (/home/leszek/projects/joystream/joystream-ws-4/cli/examples/content/cover-photo-)
Chosen storage node endpoint: http://172.18.0.1:3333/api/v1
Uploading object 1 (/home/leszek/projects/joystream/joystream-ws-4/cli/examples/content/avatar-photo)
 ›   Warning: Upload of object 0 failed: Unexpected error when trying to upload a file: 
 ›   {"type":"upload","message":"Error: Storage bag {\"dynamic\":{\"channel\":1}} is not assigned to
 ›    storage bucket 0."}
 ›   Warning: Upload of object 1 failed: Unexpected error when trying to upload a file: 
 ›   {"type":"upload","message":"Error: Storage bag {\"dynamic\":{\"channel\":1}} is not assigned to
 ›    storage bucket 0."}
 ›   Warning: Some assets were not uploaded successfully. Try reuploading them with 
 ›   content:reuploadAssets!
```

Reply:
Since #4402 supersedes this PR now, maybe we can ignore this PR and only merge #4402?